### DISCUSSION TITLE
Authenticate With Google

### DISCUSSION THREAD

**Discussion by bmmcwhirt** [mealie-recipes/mealie#4678] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Google authentication for sending E-Mail from Mealie.

### Please provide a concise description of the feature that would resolve your issue.

Beginning in January 2025 you will no longer be able to use Application passwords to authenticate to Google's SMTP servers.
[https://support.google.com/accounts/answer/6010255](https://support.google.com/accounts/answer/6010255)

I was able to create a new Application Specific password today Dec 3/4th but it fails authentication. I have other Application Specific passwords in use other places, and those do still work. This may be a time delay issue and might start working in a day, but it is confirmed that in January Application Specific passwords will cease to function.

### Please consider and list out some caveats or tradeoffs made in your design decision

Currently Mealie supports user logins with OpenID Connect. It might be possible to create an email authentication using this from the admin settings. For Docker images this would require some more work as credentials could not be stored in Env, but would have to be stored in some other way. Without looking at the code I cannot provide good advice at this time, but believe the developers probably have an idea of how to deal with this.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
I'm able to use Gmail SMTP using an app password. You might need to double check your settings.

My SMTP settings:
```
SMTP_HOST=smtp.gmail.com
SMTP_PORT=587
SMTP_FROM_NAME=Mealie
SMTP_AUTH_STRATEGY=TLS
SMTP_FROM_EMAIL=my-email@example.com
SMTP_USER=my-email@example.com
SMTP_PASSWORD=my-password
```

**Reply by bmmcwhirt** [reply]
Like I said this is a newly created app password. That might make a difference. I have also verified in security settings in my Google account that I created the app password. My account is setup with Yubikey MFA, but that should be no different than any other MFA where you create app passwords.

Even if I have a configuration issue the feature request is still valid as Google is going to start deactivating those in batches starting around the second week of January. The blog post just says January, but I've talked with people and it's not likely to start until the second week, if not the First week in February.

Also I know my email is plainly posted, but my account is locked down as tight as is possible on Google.

```
root@mealie:/# export | grep SMTP_
declare -x SMTP_AUTH_STRATEGY="TLS"
declare -x SMTP_FROM_EMAIL="bmmcwhirt@gmail.com"
declare -x SMTP_FROM_NAME="Mealie"
declare -x SMTP_HOST="smtp.gmail.com"
declare -x SMTP_PASSWORD=REDACTED
declare -x SMTP_PORT="587"
declare -x SMTP_USER="bmmcwhirt@gmail.com"
```
![Screenshot 2024-12-06 at 15-42-10 Site Settings](https://github.com/user-attachments/assets/341f01ee-3391-42ee-a196-10f21ea51d92)
